src/Sentry.AspNetCore/Extensions/HttpContextExtensions.cs

```diff
@@ -39,9 +59,16 @@ internal static class HttpContextExtensions
 
             if (!string.IsNullOrWhiteSpace(action))
             {
-                return !string.IsNullOrWhiteSpace(area)
-                    ? $"{area}.{controller}.{action}"
-                    : $"{controller}.{action}";
+                var hasArea = !string.IsNullOrWhiteSpace(area);
+                return new StringBuilder()
+                    .AppendIf(context.Request.PathBase.HasValue, context.Request.PathBase.Value?.TrimStart('/'))
+                    .AppendIf(context.Request.PathBase.HasValue, '.')
+                    .AppendIf(hasArea, area)
+                    .AppendIf(hasArea, '.')
```

Comment:
Feels like overcomplicating things having this extension to what would be 2 `if` statements with two lines within each